SNOW-1358911: Update READMEs and `CONTRIBUTING.md` with Snowpark pandas (#1601)
snowflakedb/snowpark-python: CONTRIBUTING.md

# Contributing to snowflake-snowpark-python

> Hi, thank you for taking the time to improve Snowflake's Snowpark Python or Snowpark pandas APIs!

Reviewer: ```suggestion
Hi, thank you for taking the time to improve Snowflake's Snowpark Python API!
```

The Snowpark pandas API is part of the Snowpark Python API, so I think we should leave this as is.

Author: I actually disagree with this bit b/c even though they are part of the same codebase and the pandas API relies on Python API code, it is a different API and I think deserves a callout in the title.

Reviewer: Let's get input from someone else, maybe from a snowpark-python codeowner?

Reviewer 2: Actually I just checked with Yijun and Sri offline, this is ok :)

Reviewer: @sfc-gh-jdu to be clear, the current version in this PR is ok, or the the current version in main is ok?

Reviewer 2: the current version in this PR

#### Install the library in edit mode and install its dependencies

- Create a new Python virtual environment with any Python version that we support.

>   - The Snowpark Python API supports **Python 3.8, Python 3.9, Python 3.10, and Python 3.11**. 

Author: Given the rest of the steps are the same, I think having should all the environment creation in one place is okay.

>   - Currently supported version for Snowpark Python API is **Python 3.8, Python 3.9, Python 3.10, Python 3.11**. 

Reviewer: ```suggestion
  - The Snowpark Python API supports  **Python 3.8, Python 3.9, Python 3.10, and Python 3.11**. 
```

>   - Currently supported version for Snowpark pandas API is **Python 3.9, Python 3.10, Python 3.11**. Additionally, **Modin 0.28.1** and **pandas 2.2.1** are required.

Reviewer: ```suggestion
  - The Snowpark pandas API supports **Python 3.9, Python 3.10, and Python 3.11**. Additionally, Snowpark pandas requires **Modin 0.28.1** and **pandas 2.2.1**.
```

nit: phrasing

    ```bash
    conda create --name snowpark-dev python=3.9
    ```
    
- Activate the new Python virtual environment. For example,

  ```bash
  conda activate snowpark-dev
  ```

- Go to the cloned repository root folder.
  - To install the Snowpark Python API in edit/development mode, use:

      ```bash
      python -m pip install -e ".[development, pandas]"
      ```
  - To install the Snowpark pandas API in edit/development mode, use:
      ```bash

>       python -m pip install -e ".[modin-development]"

Reviewer 3: in fact, this gives develop requirement for for snowpark python API and snowpark pandas API, right?  Can you make it clear here? and maybe add some description about the pandas library difference here?

This is little bit wired, I would expect set up modin development will require some extra library installation, but modin-development and development seems duplicate many of the required library

Author: I think it's fine to just have this section as-is, I think it would actually be less confusing for users who want to contribute to Snowpark pandas just to have this one install and not worry too much about the details.

##### Download and install VS Code

> Download and install the latest version of [VS Code](https://code.visualstudio.com/download)

Reviewer 4: The setup instructions are a bit confusing because they mix and match setup for PyCharm and VS Code. I would suggest that we don't include anything about IDE or setup here. The suggestions here are optional, basic, and anyone familiar with python development will already know how to set these things up. cc @sfc-gh-yixie

Reviewer 3: I think it is good to have some instruction for the setup for people does not familiar with python. However, the description here does seems confusing to people with a mix of both VSCode and PyCharm, we can probably clear this in a different PR

If this happens to you do not panic! Any PRs originating from a fork will fail some automated tests. This is because
forks do not have access to our repository's secrets. A maintainer will manually review your changes then kick off
the rest of our testing suite. Feel free to tag [@snowflakedb/snowpark-python-api](https://github.com/orgs/snowflakedb/teams/snowpark-python-api)
or [@snowflakedb/snowpark-pandas-api](https://github.com/orgs/snowflakedb/teams/snowpark-pandas-api)
if you feel like we are taking too long to get to your PR.

[config pycharm interpreter]: https://www.jetbrains.com/help/pycharm/configuring-python-interpreter.html
[config vscode interpreter]: https://code.visualstudio.com/docs/python/environments#_manually-specify-an-interpreter

## Snowpark pandas Folder structure

> Following tree diagram shows the high-level structure of the Snowpark pandas.

Author: I think it's fine just to have Snowpark pandas structure for Contributing guide, adding one for Snowpark Python can be done later and shouldn't block this PR

> ```bash

Reviewer 5: ```suggestion
```
```
I don't know how to suggest just 3 backticks in markdown, but you should remove the `bash` directive here.

Author: Done
